pkg/server/api/bundle/v1/service_test.go

```diff
@@ -2494,6 +2500,7 @@ func TestBatchSetFederatedBundle(t *testing.T) {
 	updatedBundle := makeValidBundle(t, federatedTrustDomain)
 	// Change the refresh hint
 	updatedBundle.RefreshHint = 120
+	updatedBundle.SequenceNumber = 42
```

Comment:
are you allowing to set a sequence number on creating and when updating a bundle? 
I think we must prevent that to happens

Reply:
Ok, so what you are saying is that even if a `newBundle` object is set with a `SequenceNumber`, it should be stored with 0?

Said with a test:

```
func (s *PluginSuite) TestCreateBundleWithSequenceNumber() {
	bundle := bundleutil.BundleProtoFromRootCA("spiffe://foo", s.cert)
	bundle.SequenceNumber = 42
	bundle, err := s.ds.CreateBundle(ctx, bundle)
	s.Require().NoError(err)
	s.Require().NotNil(bundle)

	stored, err := s.ds.FetchBundle(ctx, "spiffe://foo")
	s.Require().NoError(err)
	s.Require().NotNil(stored)
	assert.Equal(s.T(), uint64(0), stored.SequenceNumber)
}
```

In case of the method `Update`, it takes a `BundleMask` that allows you to update the `SequenceNumber`. 
Should it also NOT allow to update the `SequenceNumber`? 

If that's the case, I guess I should remove `SequenceNumber` from the `BundleMask`, right?

Reply:
Ok, 
here is the situation, 

1- In case of "server trust domain" we must not allow update sequence number, I think that is covered in case since we allow to call create or update only on federated bundles.
2- in case of federated bundles, we must allow a regular CRUD where third parties can choose any Sequence they want
3- we must allow to set the sequence on federated bundles on creation (to keep the same sequence that is on federated server database)
4-  we must prevent to set something different than 0 when creating "server" bundle

So looks like we are safe, but we'll